### ISSUE TITLE
feat: remove usage of reflection in integrations

### ISSUE THREAD

**Issue by TheJeterLP** [revanced/revanced-integrations#96]
## 🐞 Issue

Some patches (for example VideoSpeed use reflection inside the integrations. That should be changed as this decreases the user experience because reflection makes the youtube app significantly slower.

## ❗ Solution

Remove the usage of reflection and move that logic to the actual Patch.

## ❓ Motivation

Follow the general unspoken rule of java development to not use reflection at all.
Moving logic that currently uses reflection will move the time consuming logic from runtime to compile time which will increase the user experience.

**Comment by oSumAtrIX** [comment]
> because reflection makes the youtube app significantly slower

I can not reproduce this.

**Comment by TheJeterLP** [comment]
> > because reflection makes the youtube app significantly slower
> 
> I can not reproduce this.

Get an older phone and try again.

**Comment by TheJeterLP** [comment]
Ask any actual java dev and they will tell you the same

**Comment by oSumAtrIX** [comment]
No need to ask anyone else, I just have to compare it on my device. I have also tried it on an old tablet where I can not reproduce it.

**Comment by TheJeterLP** [comment]
> No need to ask anyone else, I just have to compare it on my device. I have also tried it on an old tablet where I can not reproduce it.

How did you try it?

Also maybe its not that big of an effect right now. But when more reflection continues to be used, it will obv increase. Directly access fields and methods is always faster than accessing and calling them using reflection. Thats the point

**Comment by oSumAtrIX** [comment]
> How did you try it? Also maybe its not that big of an effect right now

Yes & Yes.

**Comment by TheJeterLP** [comment]
So general rule of any development is to make the code run the fastest way you can possible do. Thats why the question about if reflection should be used or not is unneccessary. Theres only one answer for that.

**Comment by oSumAtrIX** [comment]
> So the general rule of any development is to make the code run the fastest way you can possibly do

No, this is the wrong approach. Readability and understandable code are way better than a complicated approach, especially in such cases.

**Comment by TheJeterLP** [comment]
> So general rule of any development is to make the code run the fastest way you can possibly do. 

Since we're able to achieve this goal and still be able to be failsafe using fingerprints, it should be changed to patches.

**Comment by TheJeterLP** [comment]
> > So the general rule of any development is to make the code run the fastest way you can possibly do
> 
> No, this is the wrong approach. Readability and understandable code are way better than a complicated approach, especially in such cases and where the difference is unnoticable as prevs said.

changing the integrations to not use reflection does not make the code less readablevor less understandable

**Comment by oSumAtrIX** [comment]
> changing the integrations to not use reflection does not make the code less readable or less understandable

It very much does, due to now having to resort to an abstract approach where the patch has to embed the code -  which was readable in the integrations - inside the app.

> But if the use of reflection increases it will be. 

It has to increase significantly. Reflective calls are being made thousands of times in a short time.

> I've already read comments of people complaining that revanced is slower than vanced or stock youtube was.

Placebo as there is no significant difference yet.

**Comment by TheJeterLP** [comment]
> It very much does, due to now having to resort to an abstract approach where the patch has to embed the code - which was readable in the integrations - inside the app.

I don't see how that will decrease readability. App has to be decompiled to completely understand what the patches do anyways. 

> It has to increase significantly. Reflective calls are being made thousands of times in a short time.

Which is why it should be avoided to use for us. Especially using reflection to looping through all methods in a class.

> Placebo as there is no significant difference yet.

Not for you, but you can't speak for the general user.

Also I thought that this discussion was over? You already agreed on changing the patches later.

**Comment by TheJeterLP** [comment]
Also read this: https://ncrcoe.gitbooks.io/java-for-small-teams/content/specifics/1000_do_not_use_reflection.html

**Comment by oSumAtrIX** [comment]
> I don't see how that will decrease readability.

As I said previously: 

> due to now having to resort to an abstract approach where the patch has to embed the code - which was readable in the integrations - inside the app.

> Which is why it should be avoided to use for us.

Once again:

> It has to increase significantly.

Hence it is not a high priority.

> but you can't speak for the general user

I am the general user as I have tried what the general user has tried and was not possible to reproduce. Unless no reproducible steps or something like a recording is given you can not assume that it is not a placebo. 

> Also read this

No need to. I already said:

> Readability and understandable code are way better than a complicated approach, especially in such cases and where the difference is unnoticable as prevs said.

**Comment by TheJeterLP** [comment]
I did not continue it? You did by commenting this issue? You asked to create an issue which I did.

**Comment by oSumAtrIX** [comment]
https://github.com/revanced/revanced-integrations/issues/96#issuecomment-1192738420:
> I can not reproduce this.

**Comment by TheJeterLP** [comment]
> [#96 (comment)](https://github.com/revanced/revanced-integrations/issues/96#issuecomment-1192738420):
> 
> > I can not reproduce this.

Which started the discussion again.

**Comment by oSumAtrIX** [comment]
> > [#96 (comment)](https://github.com/revanced/revanced-integrations/issues/96#issuecomment-1192738420):
> > > I can not reproduce this.
> 
> Which started the discussion again.

Please stop going off topic. The comment was related to this topic.

**Comment by AkechiShiro** [comment]
I think that if you really need such a change to be made ASAP and the maintainers are currently not in favor of it, then you should fork the patch and maintain your own flavor of Revanced Integrations @TheJeterLP 

The maintainers have made it so extensible, that anyone can take the project and extends or modify, I believe so, but I may be wrong

You will never force anyone to make a change by commenting here, even if you are right or wrong, this doesn't matter here. If you're perfectly sure that your change should make it upstream, then prove that your Revanced Integrations and your fork can out stand the official Revanced Integrations for older phone users, once you made that clear, maybe things will move upstream.

**Comment by oSumAtrIX** [comment]
Reflection has nearly no impact on performance whatsoever, but it's not really good design here and should ideally never be used. Closing though until it opposes an actual problem ala https://en.m.wikipedia.org/wiki/Ostrich_algorithm
